File: BACKEND LARAVEL/app/Models/Penduduk.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class Penduduk extends Model
{
    use HasFactory;
    protected $table="data_idv";
    protected $primarykey="id";
    protected $fillable = [
        'nomer_kk',
        'kels_id',
        'nik',
        'nama',
        'tmp_lhr',
        'tgl_lhr',
        'kelamin',
        'stat_kawin',
        'hub_kel',
        'warga_neg',
        'agama',
        'pendidikan',
        'pekerjaan',
        'ayah',
        'ibu',
        'kepala_kel',
        'no_hp',
        'domisili',
        'stat',
        'user_id'

    ];
    public $timestamps=true;

    public function keluarga(): BelongsTo
    {
        return $this->BelongsTo(keluarga::class, 'kels_id');
    }
    public function user(){
        return $this->belongsTo(user::class, 'user_id');
    }

    public static function stat($key = '')
    {
        $data = config('constants.stat');
        if ( !isset($data[$key]) )
            return $data;

        return $data[$key];
    }
    public static function pekerjaan($key = '')
    {
        $data = config('constants.pekerjaan');
        if ( !isset($data[$key]) )
            return $data;

        return $data[$key];
    }
    public static function pendidikan($key = '')
    {
        $data = config('constants.pendidikan');
        if ( !isset($data[$key]) )
            return $data;

        return $data[$key];
    }
    public static function warga_neg($key = '')
    {
        $data = config('constants.warga_negara');
        if ( !isset($data[$key]) )
            return $data;

        return $data[$key];
    }
    public static function agama($key = '')
    {
        $data = config('constants.agama');
        if ( !isset($data[$key]) )
            return $data;

        return $data[$key];
    }
    public static function kelamin($key = '')
    {
        $data = config('constants.sex');
        if ( !isset($data[$key]) )
            return $data;

        return $data[$key];
    }
    public static function stat_kawin($key = '')
    {
        $data = config('constants.status_kawin');
        if ( !isset($data[$key]) )
            return $data;

        return $data[$key];
    }
    public static function hub_kel($key = '')
    {
        $data = config('constants.shdk');
        if ( !isset($data[$key]) )
            return $data;

        return $data[$key];
    }



}

File: backend_golang/models/Penduduk.go
package models

import "time"

type Penduduk struct {
	Id         int64     `gorm:"primaryKey"`
	KelsId     int64     `json:"kels_id"`
	Nik        int64     `json:"nik"`
	Nama       string    `json:"nama"`
	TmpLahir   string    `json:"tmp_lahir"`
	TglLahir   time.Time `json:"tgl_lahir"`
	Kelamin    int8      `json:"kelamin"`
	StatKawin  int8      `json:"stat_kawin"`
	HubKel     int8      `json:"hub_kel"`
	WargaNeg   int8      `json:"warga_neg"`
	Agama      int8      `json:"agama"`
	Pendidikan int8      `json:"pendidikan"`
	Pekerjaan  int8      `json:"pekerjaan"`
	Ayah       string    `json:"ayah"`
	Ibu        string    `json:"ibu"`
	KepalaKel  string    `json:"kepala_kel"`
	NoHp       string    `json:"no_hp"`
	Domisili   int8      `json:"domisili"`
	Status     int8      `json:"status"`
	UserId     int64     `json:"user_id"`
	CreatedAt  time.Time `gorm:"type:timestamp;default:current_timestamp"`
	UpdatedAt  time.Time `gorm:"type:timestamp;default:current_timestamp on update current_timestamp"`
	User       User      `gorm:"foreignKey:UserId;references:Id;constraint:OnUpdate:CASCADE,OnDelete:SET NULL;"`
	Keluarga   Keluarga  `gorm:"foreignKey:KelsId;references:Id;constraint:OnUpdate:CASCADE,OnDelete:SET NULL;"`
}
